katdal/ms_extra.py

```diff
@@ -1026,9 +1015,11 @@ def write_dict(ms_dict, ms_name, verbose=True):
         for row_dict in sub_dict:
             if verbose:
                 print("Table %s:" % (sub_table_name,))
-            # Open table using whichever casacore library was found
-            t = open_table(ms_name, ack=verbose) if sub_table_name == 'MAIN' else \
-                open_table(os.path.join(ms_name, sub_table_name))
+            # Open main table or sub-table
+            if sub_table_name == 'MAIN':
+                t = open_table(ms_name, ack=verbose)
+            else:
+                t = open_table(os.path.join(ms_name, sub_table_name))
```

Comment:
The convention is to separate table and sub-table with ::, "WSRT.MS::ANTENNA" for e.g.

```suggestion
                t = open_table("::"join((ms_name, sub_table_name)))
```

Reply:
Thanks for the tip, done. Would you do the same inside `tables.default_ms_subtable()`?

Reply:
> Thanks for the tip, done. Would you do the same inside `tables.default_ms_subtable()`?

No, ``default_ms_subtable("SOURCE")`` will simply create a ``SOURCE`` directory in the MS. Hence the need to manually add the keyword linking the main table to it.